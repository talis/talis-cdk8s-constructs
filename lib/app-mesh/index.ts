export * from "./ingress-gateway-nlb";
export * from "./app-service";
