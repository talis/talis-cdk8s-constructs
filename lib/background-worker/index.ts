export * from "./background-worker-props";
export * from "./background-worker";
