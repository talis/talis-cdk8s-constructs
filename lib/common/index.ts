export * from "./container-props";
export * from "./env-util";
export * from "./name-util";
export * from "./statefulset-util";
export * from "./value-util";
export * from "./workload-props";
