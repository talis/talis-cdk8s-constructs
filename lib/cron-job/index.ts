export * from "./cron-job-props";
export * from "./cron-job";
