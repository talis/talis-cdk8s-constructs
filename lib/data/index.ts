export * from "./config-map";
export * from "./secret";
export * from "./factory";
