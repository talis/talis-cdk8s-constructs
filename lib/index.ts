export * from "./cron-job";
export * from "./background-worker";
export * from "./data";
export * from "./job";
export * from "./redis";
export * from "./talis-chart";
export * from "./web-service";
