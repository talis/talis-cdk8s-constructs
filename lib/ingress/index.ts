export * from "./ingress-props";
export * from "./ingress";
