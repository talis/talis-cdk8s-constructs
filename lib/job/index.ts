export * from "./job-props";
export * from "./job";
