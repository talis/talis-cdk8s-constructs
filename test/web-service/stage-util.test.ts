import { getCanaryStage } from "../../lib";

describe("stage-util", () => {
  describe("getCanaryStage", () => {
    const PROCESS_ENV = process.env;

    beforeEach(() => {
      jest.resetModules();
      process.env = { ...PROCESS_ENV };
    });

    afterEach(() => {
      process.env = PROCESS_ENV;
    });

    test("Gets name of the stage from CANARY_STAGE env var", () => {
      process.env.CANARY_STAGE = "canary";
      expect(getCanaryStage()).toBe("canary");
    });

    test("Gets name of the stage from a custom env var", () => {
      process.env.MY_STAGE = "base";
      expect(getCanaryStage("MY_STAGE")).toBe("base");
    });

    test("Throws if env var is not set", () => {
      expect(() => getCanaryStage()).toThrowErrorMatchingSnapshot();
    });

    test("Throws if env var is empty", () => {
      process.env.CANARY_STAGE = "";
      expect(() => getCanaryStage()).toThrowErrorMatchingSnapshot();
    });

    test("Throws if stage name is not valid", () => {
      process.env.CANARY_STAGE = "blue";
      expect(() => getCanaryStage()).toThrowErrorMatchingSnapshot();
    });

    test("Gets default stage if env var is not set", () => {
      expect(getCanaryStage("NOT_SET", "base")).toBe("base");
    });

    test("Gets default stage if env var is empty", () => {
      process.env.EMPTY = "";
      expect(getCanaryStage("EMPTY", "full")).toBe("full");
    });
  });
});
