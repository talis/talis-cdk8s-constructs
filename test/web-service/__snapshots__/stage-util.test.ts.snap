// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stage-util getCanaryStage Throws if env var is empty 1`] = `"Stage environment variable CANARY_STAGE is not set"`;

exports[`stage-util getCanaryStage Throws if env var is not set 1`] = `"Stage environment variable CANARY_STAGE is not set"`;

exports[`stage-util getCanaryStage Throws if stage name is not valid 1`] = `"Invalid canary stage: blue. Allowed values are base, canary, post-canary, full"`;
